src/main/java/com/rabbitmq/client/impl/recovery/AutorecoveringChannel.java

```diff
@@ -101,7 +101,7 @@ private void executeAndClean(IoTimeoutExceptionRunnable callback) throws IOExcep
             callback.run();
         } finally {
             for (String consumerTag : consumerTags) {
-                this.connection.deleteRecordedConsumer(consumerTag);
+                this.deleteRecordedConsumer(consumerTag);
```

Comment:
Iterating over the consumer tags set and removing them in `deleteRecordedConsumer(String)` causes a `ConcurrentModificationException`. Maybe moving the removal to the caller is a solution (and using an iterator and `Iterator#remove()` in the iteration). Not super-DRY, but it should work. The channel instance will be GC-ed in the case of the closing anyway.